### PR TITLE
Fixes for login 

### PR DESCRIPTION
**Pull request by meghdadyazdi**
There some issue related to login as user (and some with proposal).

### CODE REVIEW

File: mxcube3/routes/login.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def login():
             logging.getLogger("MX3.HWR").info(msg)
             res = deny_access(str(ex))
 
+        session.permanent = True
```
Comment (meguiraun) [inline mxcube3/routes/login.py:49]: just a comment. Without this line, the cookie would be valid only during the "Session", and deleted on browser closing. This causes a bug where the local storage says logged in but the absence of the cookie says the opposite, it makes the loading break. In order to reproduce (without this line), just login as normal, close the browser, open it again and go to the main page again.

The [requireAuth()](https://github.com/mxcube/mxcubeweb/blob/develop/ui/src/index.js#L42) returns before the _getLoginInfo_ finishes. This still needs to be addressed, but as long as the cookie is valid it _should_ be ok 🤔

Comment (marcus-oscarsson) [reply mxcube3/routes/login.py:49]: Aha, I wondering if we should not use this instead: https://flask-login.readthedocs.io/en/latest/#cookie-settings (REMEMBER-COOKIE)

Comment (meguiraun) [reply mxcube3/routes/login.py:49]: we should... but I did not manage to have those working 🤪...

Comment (marcus-oscarsson) [reply mxcube3/routes/login.py:49]: Alright :), its a good start. We can have a look together once you have the behavior you like :)

File: ui/src/actions/login.js [hunk 2]
```diff
@@ -149,3 +149,20 @@ export function doSignOut(navigate) {
     });
   };
 }
+
+export function signOutUser(username, navigate) {
```
Comment (marcus-oscarsson) [inline ui/src/actions/login.js]: There is already a [signout](https://github.com/mxcube/mxcubeweb/blob/81381c34c11f802d129e993c094e3bd267d9b74f/ui/src/actions/remoteAccess.js#L102), did you try to this one or did it not do what you liked ?

File: mxcube3/routes/login.py [hunk 3]
```diff
@@ -58,6 +60,18 @@ def signout():
 
         return redirect("/login", code=302)
 
+    @bp.route("/signout_user", methods=["POST"])
+    # @server.restrict
+    def signout_user():
```
Comment (marcus-oscarsson) [inline mxcube3/routes/login.py]: This is already implemented [here](https://github.com/mxcube/mxcubeweb/blob/81381c34c11f802d129e993c094e3bd267d9b74f/mxcube3/routes/ra.py#L95)

I think its about the same ?

File: mxcube3/core/components/user/usermanager.py [hunk 4]
```diff
@@ -218,8 +208,36 @@ def signout(self):
         flask_security.logout_user()
         self.emit_observers_changed()
 
+    def signout_user(self, user):
```
Comment (marcus-oscarsson) [inline mxcube3/core/components/user/usermanager.py]: You could use the method below called force_signout_user ?

**Comment by marcus-oscarsson** [comment]
Looks very nice in general, there are two questions that we have to look at. 

 - The use of permanent session (We can merge as it is, we just need to discuss the general behavior we want the)  
 -  Investigate if we can merge `signout_` and `force_signout_user`, this one would perhaps be nice to look at before we merge ?

File: ui/src/actions/login.js (hunk 2 above)
Comment (meghdadyazdi) [reply ui/src/actions/login.js]: This [signout](https://github.com/mxcube/mxcubeweb/blob/81381c34c11f802d129e993c094e3bd267d9b74f/ui/src/actions/remoteAccess.js#L102) works when you are already logged in. In our case we are not yet logged in.

File: mxcube3/routes/login.py (hunk 3 above)
Comment (meghdadyazdi) [reply mxcube3/routes/login.py]: As I said we are going to force to sign out a user (myself) while we are not yet logged in (but authenticated). Then we have to define a new method which is not @server.restrict (?)

Comment (marcus-oscarsson) [reply mxcube3/routes/login.py]: Hm, I guess we could lighten up the restriction. So far we have not needed to distinguish between logged in and authenticated, but there is perhaps a need. In that case `@server.restrict` could be made to simply check authenticated or as you mention we create a something like  `@server.restrict_authenticated`. I have to admit the the later sounds a bit odd to me. From the "restrict" point of view you are either "allowed/authenticated" or not. We could perhaps pass a parameter to `@server.restrict` like `@server.restrict(allow_authenticated = True )`. However not having any restrict could allow anyone to logout users.

Does that make sense ? 

But at at this stage why are you not logged in ?, you should be logged in but without a "selected_proposal" ?

File: ui/src/containers/SampleViewContainer.js [hunk 5]
```diff
@@ -68,6 +69,17 @@ class SampleViewContainer extends Component {
       }
     });
 
+    const phaseControl = (
```
Comment (marcus-oscarsson) [inline ui/src/containers/SampleViewContainer.js:72]: Ill merge this one for now, but there might be some discussion among the people here on where to display this. We might need to make it configurable :)

**Comment by marcus-oscarsson** [comment]
:+1: thanks !

File: ui/src/containers/SampleViewContainer.js (hunk 5 above)
Comment (meguiraun) [reply ui/src/containers/SampleViewContainer.js:72]: Oh noo! We let them fight for the location ;)
tack!

**Pull request merged**